tools/querytee/proxy.go

```diff
@@ -188,6 +238,22 @@ func NewProxy(cfg ProxyConfig, logger log.Logger, routes []Route, registerer pro
 	return p, nil
 }
 
+func validateBackendConfig(backends []ProxyBackendInterface, config map[string]*BackendConfig) error {
+	for configuredBackend := range config {
+		backendExists := false
+		for _, actualBacked := range backends {
```

Comment:
How many backends do we normally expect to have? Should we convert `backends []ProxyBackendInterface` into a map first so we only loop through both `backends` and `config` once each instead of having a nested for loop, or is it expected to be small enough not to matter?

Reply:
this only happens once on startup and backend are usually below 10. I'm not conned about perf here